src/os/source/vectorsource.js

```diff
@@ -1671,6 +1689,16 @@ os.source.Vector.prototype.createColorModel = function(opt_histogram, opt_gradie
 };
 
 
+/**
+ * Check for a color model or manually colored items in the Source
+ *
+ * @return {boolean}
+ */
+os.source.Vector.prototype.hasColors = function() {
+  return (this.colorModel != null || (this.getColoredItems() != null && this.getColoredItems().length > 0));
```

Comment:
Calling `getColoredItems()` twice here is bad as that call does an `Object.values()` on each call. Calling it once and assigning it to a local variable would be cleaner and more performant.